### DISCUSSION TITLE
Is there any way to migrate from 3rd party OAuth to a self-hosted one?

### DISCUSSION THREAD

**Discussion by Snuupy** [outline/outline#3402] [Self Hosting] [answered]
Hi,

I have a self-hosted instance using Slack's OAuth (back when that/Google OAuth were the only available ones). I see some people have had success with Keycloak or a self-hosted OIDC provider.

Is there any migration path from a self-hosted instance using Slack's OAuth to Keycloak (or some other locally hosted authentication option)? What does that look like?

**Comment by tommoor** [comment, marked as answer]
This PR will allow setting up a second auth provider if I ever get around to landing it:
https://github.com/outline/outline/pull/2986
